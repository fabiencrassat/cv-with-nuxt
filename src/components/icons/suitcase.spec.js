import { shallowMount } from '@vue/test-utils';
import Icon from './suitcase';

const factory = ({ propsData } = {}) => {
  return shallowMount(Icon, {
    mocks: {
      // Always returns the input
      $t: (i) => `$t('${i}')`,
    },
    propsData,
  });
};

describe('glasses', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    const wrapper = factory();
    // eslint-disable-next-line jest/no-truthy-falsy
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('renders properly', () => {
    expect.hasAssertions();
    const wrapper = factory();
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    const wrapper = factory({ propsData: { title: 'foo' } });
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.html()).toMatchSnapshot();
  });
});
