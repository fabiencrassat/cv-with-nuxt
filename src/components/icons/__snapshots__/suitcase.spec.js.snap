// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glasses renders properly 1`] = `
<svg viewBox="0 0 800 800" xmlns="http://www.w3.org/2000/svg" class="h-12 w-12 fill-current">
  <path d="M154.7,704h480c65.1,0,117.3-52.3,117.3-117.3v-384c0-18.1-13.9-32-32-32H554.7V160c0-53.3-42.7-96-96-96h-128   c-53.3,0-96,42.7-96,96v10.7H69.3c-18.1,0-32,13.9-32,32v384C37.3,651.7,89.6,704,154.7,704z M298.7,160c0-18.1,13.9-32,32-32h128   c18.1,0,32,13.9,32,32v10.7h-192V160z M101.3,234.7H688V320c0,22.4-13.9,42.7-35.2,50.1l0,0l-240,85.3l0,0l0,0   c-11.7,4.3-24.5,4.3-35.2,0l0,0l0,0l-240-85.3l0,0c-21.3-7.5-35.2-27.7-35.2-50.1v-85.3H101.3z M101.3,424.5   c4.3,2.1,9.6,4.3,13.9,6.4l0,0l240,85.3l0,0l0,0l0,0l0,0c12.8,4.3,25.6,6.4,39.5,6.4c13.9,0,26.7-2.1,39.5-6.4l0,0l0,0l0,0l0,0   l240-85.3l0,0c5.3-2.1,9.6-4.3,13.9-6.4v162.1c0,29.9-23.5,53.3-53.3,53.3h-480c-29.9,0-53.3-23.5-53.3-53.3V424.5z"></path>
  <title>$t('title')</title>
</svg>
`;

exports[`glasses renders properly with props 1`] = `
<svg viewBox="0 0 800 800" xmlns="http://www.w3.org/2000/svg" class="h-12 w-12 fill-current">
  <path d="M154.7,704h480c65.1,0,117.3-52.3,117.3-117.3v-384c0-18.1-13.9-32-32-32H554.7V160c0-53.3-42.7-96-96-96h-128   c-53.3,0-96,42.7-96,96v10.7H69.3c-18.1,0-32,13.9-32,32v384C37.3,651.7,89.6,704,154.7,704z M298.7,160c0-18.1,13.9-32,32-32h128   c18.1,0,32,13.9,32,32v10.7h-192V160z M101.3,234.7H688V320c0,22.4-13.9,42.7-35.2,50.1l0,0l-240,85.3l0,0l0,0   c-11.7,4.3-24.5,4.3-35.2,0l0,0l0,0l-240-85.3l0,0c-21.3-7.5-35.2-27.7-35.2-50.1v-85.3H101.3z M101.3,424.5   c4.3,2.1,9.6,4.3,13.9,6.4l0,0l240,85.3l0,0l0,0l0,0l0,0c12.8,4.3,25.6,6.4,39.5,6.4c13.9,0,26.7-2.1,39.5-6.4l0,0l0,0l0,0l0,0   l240-85.3l0,0c5.3-2.1,9.6-4.3,13.9-6.4v162.1c0,29.9-23.5,53.3-53.3,53.3h-480c-29.9,0-53.3-23.5-53.3-53.3V424.5z"></path>
  <title>foo</title>
</svg>
`;
